--- origsrc/webkitgtk-2.0.4/Source/ThirdParty/ANGLE/src/compiler/osinclude.h	2013-01-21 10:37:26.000000000 -0600
+++ src/webkitgtk-2.0.4/Source/ThirdParty/ANGLE/src/compiler/osinclude.h	2014-07-04 15:53:20.588790400 -0500
@@ -18,7 +18,7 @@
       defined(__FreeBSD__) || defined(__OpenBSD__) || \
       defined(__sun) || defined(ANDROID) || \
       defined(__GLIBC__) || defined(__GNU__) || \
-      defined(__QNX__)
+      defined(__QNX__) || defined(__CYGWIN__)
 #define ANGLE_OS_POSIX
 #else
 #error Unsupported platform.
@@ -43,6 +43,9 @@
 #if defined(ANGLE_OS_WIN)
 typedef DWORD OS_TLSIndex;
 #define OS_INVALID_TLS_INDEX (TLS_OUT_OF_INDEXES)
+#elif defined(__CYGWIN__)
+typedef pthread_key_t OS_TLSIndex;
+#define OS_INVALID_TLS_INDEX (NULL)
 #elif defined(ANGLE_OS_POSIX)
 typedef pthread_key_t OS_TLSIndex;
 #define OS_INVALID_TLS_INDEX (static_cast<OS_TLSIndex>(-1))
--- origsrc/qtwebkit-97c4a80a1282c8c3eaa343011286b76fd4838c5f/Source/ThirdParty/ANGLE/src/compiler/ossource_posix.cpp	2017-06-30 02:53:10.000000000 -0500
+++ src/qtwebkit-97c4a80a1282c8c3eaa343011286b76fd4838c5f/Source/ThirdParty/ANGLE/src/compiler/ossource_posix.cpp	2017-10-24 09:56:22.133703700 -0500
@@ -25,7 +25,7 @@ OS_TLSIndex OS_AllocTLSIndex()
     //
     if ((pthread_key_create(&pPoolIndex, NULL)) != 0) {
         assert(0 && "OS_AllocTLSIndex(): Unable to allocate Thread Local Storage");
-        return false;
+        return OS_INVALID_TLS_INDEX;
     }
     else {
         return pPoolIndex;
