--- origsrc/qtwebkit-opensource-src-5.3.2/Source/WebCore/Target.pri	2014-09-11 05:48:19.000000000 -0500
+++ src/qtwebkit-opensource-src-5.3.2/Source/WebCore/Target.pri	2015-03-29 14:33:28.682148200 -0500
@@ -4310,11 +4310,6 @@ use?(GRAPHICS_SURFACE) {
     }
 }
 
-build?(qttestsupport) {
-    HEADERS += platform/qt/QtTestSupport.h
-    SOURCES += platform/qt/QtTestSupport.cpp
-}
-
 ALL_IN_ONE_SOURCES += \
     accessibility/AccessibilityAllInOne.cpp \
     inspector/InspectorAllInOne.cpp \
--- origsrc/qtwebkit-opensource-src-5.3.2/Source/WebCore/WebCore.pri	2014-09-11 05:48:12.000000000 -0500
+++ src/qtwebkit-opensource-src-5.3.2/Source/WebCore/WebCore.pri	2015-03-29 14:34:54.182745300 -0500
@@ -302,6 +302,7 @@ mac {
 }
 unix:!mac:*-g++*:QMAKE_CXXFLAGS += -fdata-sections
 unix:!mac:*-g++*:QMAKE_LFLAGS += -Wl,--gc-sections
+cygwin*:QMAKE_LFLAGS += -Wl,--exclude-libs=libJavaScriptCore.a:libANGLE.a:libWebCore.a
 linux*-g++*:QMAKE_LFLAGS += $$QMAKE_LFLAGS_NOUNDEF
 
 enable_fast_mobile_scrolling: DEFINES += ENABLE_FAST_MOBILE_SCROLLING=1
--- origsrc/qtwebkit-opensource-src-5.3.2/Source/WebKit/WebKit1.pro	2014-09-11 05:48:12.000000000 -0500
+++ src/qtwebkit-opensource-src-5.3.2/Source/WebKit/WebKit1.pro	2015-03-29 14:34:06.026125900 -0500
@@ -121,4 +121,8 @@ enable?(VIDEO) {
     }
 }
 
+build?(qttestsupport) {
+    HEADERS += $$PWD/../WebCore/platform/qt/QtTestSupport.h
+    SOURCES += $$PWD/../WebCore/platform/qt/QtTestSupport.cpp
+}
 
