--- origsrc/qtwebkit-opensource-src-5.4.1/Source/WTF/wtf/ExportMacros.h	2015-04-27 18:45:37.939492000 -0500
+++ src/qtwebkit-opensource-src-5.4.1/Source/WTF/wtf/ExportMacros.h	2015-04-28 16:42:44.281063000 -0500
@@ -48,7 +48,7 @@
 #define WTF_INTERNAL
 #endif
 
-#if OS(WINDOWS)
+#if OS(WINDOWS) || OS(CYGWIN)
 
 #define WTF_EXPORT_DECLARATION __declspec(dllexport)
 #define WTF_IMPORT_DECLARATION __declspec(dllimport)
@@ -89,7 +89,7 @@
 
 #else // !USE(EXPORT_MACROS)
 
-#if OS(WINDOWS) && !COMPILER(GCC)
+#if OS(WINDOWS) || OS(CYGWIN)
 #if defined(BUILDING_WTF) || defined(STATICALLY_LINKED_WITH_WTF)
 #define WTF_EXPORTDATA __declspec(dllexport)
 #else
--- origsrc/qtwebkit-opensource-src-5.4.1/Source/WebCore/platform/PlatformExportMacros.h	2015-02-16 22:57:06.000000000 -0600
+++ src/qtwebkit-opensource-src-5.4.1/Source/WebCore/platform/PlatformExportMacros.h	2015-04-28 09:23:41.739985100 -0500
@@ -48,7 +48,7 @@
 
 #else // !USE(EXPORT_MACROS)
 
-#if OS(WINDOWS) && !COMPILER(GCC)
+#if OS(WINDOWS) || OS(CYGWIN)
 
 #if defined(WEBCORE_IS_LINKED_IN_SAME_BINARY)
 #define WEBKIT_EXPORTDATA __declspec(dllexport)
